dulwich/contrib/paramiko_vendor.py

```diff
@@ -97,7 +97,7 @@ def read(self, n=None):
 
         # Closed socket
         if not data:
```

Comment:
Why is this code being hit at all by the tests?

Reply:
```
channel.send(b'stdout\n')
self.assertEqual(b'stdout\n', con.read(4096))
```